### DISCUSSION TITLE
Unable to build windows2019 image

### DISCUSSION THREAD

**Discussion by tedm13** [actions/virtual-environments#5780] [General questions] [answered]
Hello,

I've been trying to build a windows 2019 image for the past few days but keep getting failed builds. 

This is the command that I initiate to start the build process. I've been running this on my machine since I read that using Azure DevOps has a time limit of 6 hours and won't work.

`GenerateResourcesAndImage -SubscriptionId {YourSubscriptionId} -ResourceGroupName "rg-vm-images-westus2" -ImageGenerationRepositoryRoot "$pwd" -ImageType windows2019 -AzureLocation "West US2"`

After 5ish hours I originally go this error:
    vhd:
    vhd: Chocolatey installed 0/1 packages. 1 packages failed.
    vhd:  See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
    vhd:
    vhd: Failures
    vhd:  - postgresql (exited 1) - postgresql not installed. An error occurred during installation:
    vhd:  Unable to resolve dependency 'postgresql14 (= 14.4.1)'.
    vhd: Running [#3]: choco install postgresql -y --params '/Password:root /NoPath' --params-global --version=14.4.1
    vhd: Chocolatey v1.1.0
    vhd: Installing the following packages:
    vhd: postgresql
    vhd: By installing, you accept licenses for the packages.
    vhd: postgresql not installed. An error occurred during installation:
    vhd:  Unable to resolve dependency 'postgresql14 (= 14.4.1)'.
    vhd: postgresql package files install failed with exit code 1. Performing other installation steps.
    vhd: The install of postgresql was NOT successful.
    vhd: postgresql not installed. An error occurred during installation:
    vhd:  Unable to resolve dependency 'postgresql14 (= 14.4.1)'.
    vhd:
    vhd: Chocolatey installed 0/1 packages. 1 packages failed.
    vhd:  See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
    vhd:
    vhd: Failures
    vhd:  - postgresql (exited 1) - postgresql not installed. An error occurred during installation:
    vhd:  Unable to resolve dependency 'postgresql14 (= 14.4.1)'.
    vhd: Running [#4]: choco install postgresql -y --params '/Password:root /NoPath' --params-global --version=14.4.1
    vhd: Chocolatey v1.1.0
    vhd: Installing the following packages:
    vhd: postgresql
    vhd: By installing, you accept licenses for the packages.
    vhd: postgresql not installed. An error occurred during installation:
    vhd:  Unable to resolve dependency 'postgresql14 (= 14.4.1)'.
    vhd: postgresql package files install failed with exit code 1. Performing other installation steps.
    vhd: The install of postgresql was NOT successful.
    vhd: postgresql not installed. An error occurred during installation:
    vhd:  Unable to resolve dependency 'postgresql14 (= 14.4.1)'.
    vhd:
    vhd: Chocolatey installed 0/1 packages. 1 packages failed.
    vhd:  See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
    vhd:
    vhd: Failures
    vhd:  - postgresql (exited 1) - postgresql not installed. An error occurred during installation:
    vhd:  Unable to resolve dependency 'postgresql14 (= 14.4.1)'.
    vhd: Could not install postgresql after 5 attempts
==> vhd: Script exited with non-zero exit status: 1.Allowed exit codes are: [0]
==> vhd: Step "StepProvision" failed
==> vhd: [c] Clean up and exit, [a] abort without cleanup, or [r] retry step (build may fail even if retry succeeds)?

So I ended up removing a few lines from the windows2019.json...
                "{{ template_dir }}/scripts/Installers/Install-AndroidSDK.ps1",
                "{{ template_dir }}/scripts/Installers/Install-PHP.ps1",
                "{{ template_dir }}/scripts/Installers/Install-Rust.ps1",
                "{{ template_dir }}/scripts/Installers/Install-MysqlCli.ps1",
                "{{ template_dir }}/scripts/Installers/Install-SQLPowerShellTools.ps1",
                "{{ template_dir }}/scripts/Installers/Install-AzureCosmosDbEmulator.ps1",
                "{{ template_dir }}/scripts/Installers/Install-PostgreSQL.ps1",

but unfortunately met with other errors during the test process so tried again but now I removed the last testing step
        {
            "type": "powershell",
            "inline": [
                "pwsh -File '{{user `image_folder`}}\\SoftwareReport\\SoftwareReport.Generator.ps1'"
            ],
            "environment_vars": [
                "IMAGE_VERSION={{user `image_version`}}"
            ]
        },
        {
            "type": "powershell",
            "inline": [
                "if (-not (Test-Path C:\\InstalledSoftware.md)) { throw 'C:\\InstalledSoftware.md not found' }"
            ]
        },
        {
            "type": "file",
            "source": "C:\\InstalledSoftware.md",
            "destination": "{{ template_dir }}/Windows2019-Readme.md",
            "direction": "download"
        },

but still ran into another issue with tests. 

==> vhd: Test run has failed
    vhd: Containers            : {[+] C:\image\Tests\Apache.Tests.ps1, [+] C:\image\Tests\BizTalk.Tests.ps1, [+]
    vhd:                         C:\image\Tests\Browsers.Tests.ps1, [+] C:\image\Tests\ChocoPackages.Tests.ps1...}
    vhd: Result                : Failed
    vhd: FailedCount           : 2
    vhd: FailedBlocksCount     : 0
    vhd: FailedContainersCount : 0
    vhd: PassedCount           : 441
    vhd: SkippedCount          : 2
    vhd: NotRunCount           : 0
    vhd: TotalCount            : 445
    vhd: Duration              : 00:04:37.1270286
    vhd: Executed              : True
    vhd: ExecutedAt            : 6/20/2022 7:46:25 PM
    vhd: Version               : 5.3.3
    vhd: PSVersion             : 5.1.17763.2931
    vhd: PSBoundParameters     : {[Configuration, PesterConfiguration]}
    vhd: Plugins               :
    vhd: PluginConfiguration   :
    vhd: PluginData            :
    vhd: Configuration         : PesterConfiguration
    vhd: DiscoveryDuration     : 00:00:05.8638163
    vhd: UserDuration          : 00:04:23.3145412
    vhd: FrameworkDuration     : 00:00:07.9486711
    vhd: Failed                : {[-] Azure Cosmos DB Emulator install location registry key exists, [-] Azure Cosmos DB
    vhd:                         Emulator exe file exists}
    vhd: FailedBlocks          : {}
    vhd: FailedContainers      : {}
    vhd: Passed                : {[+] Apache, [+] Apache service is stopped, [+] Apache service is disabled, [+] BizTalk
    vhd:                         Registry Check...}
    vhd: Skipped               : {[!] vcredist_140, [!] Verifies 17763 SDK is installed}
    vhd: NotRun                : {}
    vhd: Tests                 : {[+] Apache, [+] Apache service is stopped, [+] Apache service is disabled, [+] BizTalk
    vhd:                         Registry Check...}
    vhd: CodeCoverage          :
    vhd:
==> vhd: At C:\Program Files\WindowsPowerShell\Modules\ImageHelpers\TestsHelpers.ps1:73 char:9
==> vhd: +         throw "Test run has failed"
==> vhd: +         ~~~~~~~~~~~~~~~~~~~~~~~~~~~
==> vhd:     + CategoryInfo          : OperationStopped: (Test run has failed:String) [], RuntimeException
==> vhd:     + FullyQualifiedErrorId : Test run has failed
==> vhd:
==> vhd: Script exited with non-zero exit status: 1.Allowed exit codes are: [0]
==> vhd: Step "StepProvision" failed

Is there something I need to do differently to get this to build successfully? Is there a way I can salvage this build or do I need to start over? Thanks for looking

**Comment by miketimofeev** [comment, marked as answer]
@tedm13 could you try building one more time using the full template and pulling the latest changes from main?

**Reply by tedm13** [reply]
![Capture](https://user-images.githubusercontent.com/78162200/175041222-075505b9-b9bf-4f62-a968-013ecb4558eb.PNG)

Received a 500 error when trying to download the ghc package. I am trying again now but it's a bit frustrating that I can't retry when something like this happens and requires me to do a full rebuild. When I select retry as an option it errors at account creation and continues until it tries to resize the partition it looks like. This is the error I get when I try to retry a failed build

==> vhd: + Resize-Partition -DriveLetter $driveLetter -Size $size.SizeMax
==> vhd: + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
==> vhd:     + CategoryInfo          : NotSpecified: (StorageWMI:ROOT/Microsoft/.../MSFT_Partition) [Resize-Partition], CimExce
==> vhd:    ption
==> vhd:     + FullyQualifiedErrorId : StorageWMI 4097,Resize-Partition
==> vhd:
==> vhd: Script exited with non-zero ex

**Comment by tedm13** [comment]
Looks like pulling master and just retrying after the 500 error worked. Built the image and using the VM image. Thank you!
